x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_notify_when.tsx

```diff
@@ -0,0 +1,238 @@
+      }
+    ),
+    'data-test-subj': 'onActionGroupChange',
+    dropdownDisplay: (
+      <>
+        <strong>
+          <FormattedMessage
+            defaultMessage="On status changes"
+            id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActionGroupChange.label"
+          />
+        </strong>
+        <EuiText size="s" color="subdued">
+          <p>
+            <FormattedMessage
+              defaultMessage="Actions run if the alert status changes."
+              id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActionGroupChange.description"
+            />
+          </p>
+        </EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'onActiveAlert',
+    inputDisplay: i18n.translate(
+      'xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActiveAlert.display',
+      {
+        defaultMessage: 'On check intervals',
+      }
+    ),
+    'data-test-subj': 'onActiveAlert',
+    dropdownDisplay: (
+      <>
+        <strong>
+          <FormattedMessage
+            defaultMessage="On check intervals"
+            id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActiveAlert.label"
+          />
+        </strong>
+        <EuiText size="s" color="subdued">
+          <p>
+            <FormattedMessage
+              defaultMessage="Actions run if rule conditions are met."
+              id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActiveAlert.description"
+            />
+          </p>
+        </EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'onThrottleInterval',
+    inputDisplay: i18n.translate(
+      'xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.display',
+      {
+        defaultMessage: 'On custom action intervals',
+      }
+    ),
+    'data-test-subj': 'onThrottleInterval',
+    dropdownDisplay: (
+      <>
+        <strong>
+          <FormattedMessage
+            defaultMessage="On custom action intervals"
+            id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.label"
+          />
+        </strong>
+        <EuiText size="s" color="subdued">
+          <p>
+            <FormattedMessage
+              defaultMessage="Actions run if rule conditions are met."
+              id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.description"
+            />
+          </p>
+        </EuiText>
+      </>
+    ),
+  },
+];
+
+interface RuleNotifyWhenProps {
+  frequency: RuleAction['frequency'];
+  throttle: number | null;
+  throttleUnit: string;
+  onNotifyWhenChange: (notifyWhen: RuleNotifyWhenType) => void;
+  onThrottleChange: (throttle: number | null, throttleUnit: string) => void;
+}
+
+export const ActionNotifyWhen = ({
+  frequency = DEFAULT_FREQUENCY,
+  throttle,
+  throttleUnit,
+  onNotifyWhenChange,
+  onThrottleChange,
+}: RuleNotifyWhenProps) => {
+  const [ruleThrottle, setRuleThrottle] = useState<number>(throttle || 1);
```

Comment:
I am wondering if we really need this intermediate state here, I think we can use directly `frequency` and `throttle` directly with their callback.